src/content.js

```diff
@@ -186,6 +186,19 @@ var content = (function() {
       }
     },
 
+    getParentsAndSelf: function(node, tagName) {
+      var parents = [];
+      while (node) {
+        if (node.nodeType === nodeType.elementNode) {
+          if (!tagName || node.nodeName === tagName.toUpperCase()) {
+            parents.push(node);
+          }
+        }
+        node = node.parentNode;
+      }
+      return parents;
+    },
```

Comment:
@peyerluk, would it be possible to merge only this function if you don’t want to merge the patch below? I ask because it comes in handy in other contexts as well, e.g. avoiding `<em>` inside of a `<strong>` or others. 

To address @marcbachmann’s [concern](https://github.com/livingdocsIO/editable.js/pull/99/files#r26464779) perhaps a flag that stops traversal at a given node would be an agreeable compromise?

Lastly, I’d probably rename the function to `getAncestorsAndSelf()` to be more in line with the XPath axis [ancestor-or-self](https://developer.mozilla.org/en-US/docs/Web/XPath/Axes/ancestor-or-self).

Reply:
This function is a bit of an odd duck currently as this is the only method that is meant to collect info outside of the host element.

It would need some polishing before we can merge it.

Would this method here help your use case? Or are you more interested in knowing which tags affect a cursor or selection within the host? For that we have the method `getTags()` in this file.

Reply:
> Would this method here help your use case?

@peyerluk, I think I used `getTagNames()` from PR https://github.com/livingdocsIO/editable.js/pull/198 in its stead.